src/main/java/org/dice_research/opal/civet/metrics/AvailabilityOfLicensesMetric.java

```diff
@@ -0,0 +1,175 @@
+import org.apache.jena.rdf.model.Resource;
+import org.apache.jena.rdf.model.ResourceFactory;
+import org.apache.jena.rdf.model.SimpleSelector;
+import org.apache.jena.rdf.model.Statement;
+import org.apache.jena.rdf.model.StmtIterator;
+import org.apache.jena.vocabulary.DCAT;
+import org.apache.jena.vocabulary.DCTerms;
+import org.apache.jena.vocabulary.RDF;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.dice_research.opal.civet.Metric;
+import org.dice_research.opal.common.vocabulary.Opal;
+
+/**
+ * The AvailabilityOfLicensesMetric provides a rating to a dataset
+ * based on the number of available licenses/rights in a dataset.
+ * 
+ * @author Gourab Sahu
+ */
+public class AvailabilityOfLicensesMetric implements Metric {
+
+	public static boolean isValidLicenseURL(String LicenseURL) {
+		/*
+		 * new URL tries to create a new URL with provided license URL. If invalid
+		 * License URL then catch exception and return false.
+		 */
+		try {
+			new URL(LicenseURL).toURI();
+			return true;
+		}
+
+		catch (Exception e) {
+			return false;
+		}
+	}
+
+	private static final Logger LOGGER = LogManager.getLogger();
+	private static final String DESCRIPTION = 
+			 "If dataset has rights/license info then award 5 stars "
+			+ "Else If dataset has no rights/license but all distributions in dataset have licenses/rights then give 5 star"
+			+ "Else If dataset has no rights/license but less than 100% and more than or equal to 75% dataset's distribution has then 4 stars are awarded"
+			+ "Else If dataset has no rights/license but less than 75% and more than or equal to 50% dataset's distribution has then 3 stars are awarded"
+			+ "Else If dataset has no rights/license but less than 50% and more than or equal to 25% dataset's distribution has then 2 stars are awarded"
+			+ "Else If dataset has no rights/license but less than 25% and more than  0% dataset's distribution has then 1 star is awarded"
+			+ "Else if no License at all then 0 star is awarded.";
+
+	@Override
+	public Integer compute(Model model, String datasetUri) throws Exception {
+
+		LOGGER.info("Processing dataset " + datasetUri);
+
+		Resource dataset = ResourceFactory.createResource(datasetUri);
+
+		// NodeIterator nodeIterator = model.listObjectsOfProperty(dataset,
+		// DCAT.keyword);
+
+		// For Score calculation
+		int TotalDistributions = 0;
+		int TotalDistributionsWithLicense = 0;
+		int TotalDistributionsWithRights = 0;
+
+		// score to return
+		int score = 0;
+
+		/*
+		 * For dcat:dataset, we will first check if license is given or not. If not
+		 * check anything with 'right'.
+		 */
+		int NumberOfLicensesInDataset = 0;
+		NodeIterator LicensesIterator = model.listObjectsOfProperty(dataset,DCTerms.license);
+		NodeIterator RightsIterator = model.listObjectsOfProperty(dataset,DCTerms.rights);
+		if (LicensesIterator.hasNext()) {
+           RDFNode licensObject = LicensesIterator.nextNode();
```

Comment:
indentation is needed

Reply:
Where? I guess already provided.

Reply:
line nuber 81 , though you can just ignore this .